larq_compute_engine/core/bitpack.h

```diff
@@ -33,13 +33,15 @@ inline void bitpack_bitfield_quantized(const TIn* in, TBitpacked* out,
                                        const std::int32_t zero_point) {
   ruy::profiler::ScopeLabel label(
       "Bitpack bitfield (quantized input, unoptimised)");
-  *out = 0;
+  TBitpacked result = 0;
+  TBitpacked bit = TBitpacked(1);
   for (size_t i = 0; i < bitpacking_bitwidth; ++i) {
     // Note: uint8 to int32 will set the top 24 bits to 0
     //        int8 to int32 will set the top 24 bits to the int8 sign bit
-    if (static_cast<std::int32_t>(in[i]) < zero_point)
-      *out |= (TBitpacked(1) << i);
+    if (static_cast<std::int32_t>(in[i]) < zero_point) result |= bit;
+    bit <<= 1;
```

Comment:
I think if the loop is unrolled you can do shift + or in a single cycle can't you (because the shift is constant)?

Reply:
(Which is to say, the existing line `result |= (TBitpacked(1) << i)` could be done in a single cycle if `i` is constant?

Reply:
Good point, by unrolling it into 32 pieces the shift will become a constant and can then be incorporated into the `orr`.
I'll benchmark a few implementations soon.

Reply:
Benchmarking showed that the "bitfield method" was fastest, so I went with that.